api/src/backend/api/v1/serializers.py

```diff
@@ -2064,6 +2070,128 @@ def update(self, instance, validated_data):
         return super().update(instance, validated_data)
 
 
+# Processors
+
+
+class ProcessorSerializer(RLSSerializer):
+    """
+    Serializer for the Processor model.
+    """
+
+    configuration = ProcessorConfigField()
+
+    class Meta:
+        model = Processor
+        fields = [
+            "id",
+            "inserted_at",
+            "updated_at",
+            "processor_type",
+            "configuration",
+            "url",
+        ]
+
+
+class ProcessorCreateSerializer(RLSSerializer, BaseWriteSerializer):
+    configuration = ProcessorConfigField(required=True)
+
+    class Meta:
+        model = Processor
+        fields = [
+            "inserted_at",
+            "updated_at",
+            "processor_type",
+            "configuration",
+        ]
+        extra_kwargs = {
+            "inserted_at": {"read_only": True},
+            "updated_at": {"read_only": True},
+        }
+        validators = [
+            UniqueTogetherValidator(
+                queryset=Processor.objects.all(),
+                fields=["processor_type", "configuration"],
+                message="A processor with the same type already exists.",
+            )
+        ]
```

Comment:
Is `configuration` expected as well?

Reply:
Not sure about this. The uniqueness constraint is processor and tenat right? At least for Mutelist that's the constraint but not sure at this point what'd be for others.

Reply:
Removed the `configuration` field from the uniqueness constraint in [`00e9d16` (#7993)](https://github.com/prowler-cloud/prowler/pull/7993/commits/00e9d165574b9a294b437756a68d216963cb2f74)

![Screenshot 2025-07-08 at 17 03 14](https://github.com/user-attachments/assets/2102f668-1cf0-4db6-ac68-1e627d062a36)

So, each processor is unique per tenant and type.